main.rb

```diff
@@ -1 +1,173 @@
-puts 'This is template of Ruby for projects in Microverse'
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength, Style/ClassEqualityComparison
+module Enumerable
+  # my_each method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    to_a.size.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    to_a.size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  # my_select method
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    to_a.size.times do |i|
+      arr.push(to_a[i]) if yield to_a[i]
+    end
+    return arr.to_h if is_a? Hash
+
+    arr
+  end
+
+  def my_inject(arg1 = nil, arg2 = nil)
+    if arg1.is_a?(Symbol) && !arg2
+      base = to_a[0]
+      1.upto(to_a.length - 1) { |item| base = base.send(arg1, to_a[item]) }
+    elsif !arg1.is_a?(Symbol) && arg2.is_a?(Symbol)
+      base = arg1
+      0.upto(to_a.length - 1) { |item| base = base.send(arg2, to_a[item]) }
+    elsif block_given? && arg1
+      base = arg1
+      to_a.my_each { |val| base = yield(base, val) }
+    elsif block_given? && !arg1
+      base = to_a[0]
+      1.upto(to_a.length - 1) { |item| base = yield(base, to_a[item]) }
+    elsif !block_given? && !arg1
+      raise LocalJumpError
+    else
+      raise LocalJumpError, 'No block given'
+    end
+    base
+  end
+
+  def my_all?(args = nil)
+    unless block_given?
+      if args.instance_of?(Regexp)
+        to_a.my_each { |var| return false unless args.match(var) }
+        return true
+      elsif args
+        to_a.my_each { |var| return false unless my_instance_of(args, var) }
+      end
+      to_a.my_each { |var| return false unless var }
+      return true
+    end
+    to_a.size.times { |i| return false unless yield to_a[i] }
+
+    true
+  end
+
+  # my_any? Method
+  def my_any?(args = nil)
+    unless block_given?
+      if args.instance_of?(Regexp)
+        to_a.my_each { |var| return true if args.match(var) }
+        return false
+      elsif args
+        to_a.my_each { |var| return true if my_instance_of(args, var) }
+      end
+
+      to_a.my_each { |var| return true if var == true }
+      return false
+    end
+    to_a.my_each { |var| return true if yield var }
+    false
+  end
```

Comment:
- [ ] When a pattern other than Regex or a Class is given, it should return true if any of the collection matches the pattern.